src/js/page_sections/Navmenu.js

```diff
@@ -19,104 +20,109 @@ class Navmenu extends Component {
   }
 
   componentDidUpdate() {
     if (this.props.isOpen) {
-      this.focusOnClose()
+      this.focusOnClose();
     }
   }
 
   getCurrentPath = () => {
     // this is dependent on react-router. We may want to use
     // `window.location.href` instead.
-    return this.props.location.pathname
-  }
-
-  getMenuItemClassName = (path) => {
-    const pathname = this.getCurrentPath()
-    return pathname === path ? 'usa-current' : ''
+    return this.props.location.pathname;
   }
 
   getOverlayClassName = () => {
     let className = `coa-Navmenu__overlay`;
     if (this.props.isOpen) {
-      className = `${className} ${className}--open`
+      className = `${className} ${className}--open`;
     }
-    return className
+    return className;
   }
 
-  getParentMenuItemClassName = (topic) => {
-    const pathname = this.getCurrentPath()
+  getParentMenuItemClassName = (parentLink) => {
+    const currentPath = this.getCurrentPath();
+    const servicePaths = parentLink.services.edges
+      ? parentLink.services.edges.map(({ node:serviceLink }) => '/service/' + serviceLink.slug)
+      : [];
 
-    const currentId = Number(pathname.split('/')[2]) // this is brittle
-    const childrenIds = topic.children.map((child) => child.id)
-    const isActive = includes(childrenIds, currentId)
+    const isActive = includes(servicePaths, currentPath);
 
-    return isActive ? 'usa-current' : ''
+    return isActive ? 'usa-current' : '';
+  }
+
+  getMenuItemClassName = (path) => {
+    const currentPath = this.getCurrentPath();
+    return currentPath === path ? 'usa-current' : '';
   }
 
   render() {
-    return (
+
+    if (!this.state.isLoaded) return 'loading...';
+
+    const { edges: parentLinks = [] } = this.state.data;
+
+    return parentLinks.length && (
       <div className="usa-grid-full">
-        <nav role="navigation" className={this.menuClassName()}>
+        <nav className={this.menuClassName()}>
           <button className="coa-Navmenu__close-btn" onClick={this.props.toggleMenu} ref="closeTrigger" tabIndex="0">
             <CloseSVG size="40" />
           </button>
           <ul className="usa-sidenav-list">
             <li>
               <Link to="/" className={this.getMenuItemClassName('/')}>Home</Link>
             </li>
-            { this.state.nav && this.state.nav.map((parent) => {
-              return (
-                <li>
-                  <Link to="/services" className={this.getParentMenuItemClassName(parent)}>
-                    {parent.parentTitle}
-                  </Link>
+
+        {
+          parentLinks.map(({ node: parentLink }) => {
+
+            let { edges: serviceLinks = [] } = parentLink.services;
+
+            return (
+              <li key={parentLink.id}>
+                <Link to="/services" className={this.getParentMenuItemClassName(parentLink)}>
+                  { parentLink.text }
+                </Link>
+
+                { serviceLinks.length && (
```

Comment:
Should we stay consistent with naming and call this `childLinks` for `parentLinks`?

Reply:
instead of `serviceLinks`?

Reply:
I kept this specific, since the API returns specifically service links and in the future might return dept links and/or event links etc. We're also currently being specific on how we build paths (/service/slug/) -- so I figured specifying would be best since event and dept links would be built differently.